Tests/ParsingTests/ConditionalTests.swift

```diff
@@ -13,13 +13,13 @@ final class ConditionalTests: XCTestCase {
 
   func testFirst() {
     var input = "42 Hello, world!"[...]
-    XCTAssertEqual(true, try parser.parse(&input))
+    XCTAssertEqual(true, try parser.parse(&input.utf8))
```

Comment:
Another consequence of this change is that using the `inout` interfaces now goes through `.utf8`. We think this is probably OK, and uncommon outside the library's own tests, since you will usually invoke the non-`inout` version.

We could consider an overload, but we don't think it's necessary...

Reply:
Seems like it'll impact the tests of parsers build on top of the Parsing package also, won't it?  If so, that means it impacts _everyone_ who's going to be using this package, because they're all going to write copious tests, of course! ;)

Reply:
Ok.  Remove the "seems" from that comment.  Migrated a small test project I have over to this PR branch and, yes, _all_ my tests need to be modified.  So this seems likely to impact test writing users of this package.

The rest of the changes to incorporate this PR were relatively minor.  Changing all the tests was the largest change required to migrate over (not hard of course, but seemed like busywork).

Reply:
The other place this is going to trip up new users is when they're just playing around in a playground because that's where the `inout` version of parse is often used (as you do in your videos).  Given that part of the goal of this PR is to make it easier for new users, would be better not to require they figure out to add `.utf8` (and wonder "why?").

Reply:
> Ok. Remove the "seems" from that comment. Migrated a small test project I have over to this PR branch and, yes, _all_ my tests need to be modified. So this seems likely to impact test writing users of this package.

@haikusw Thanks for trying this branch out!

Are you asserting against the input after it's been consumed? If so, can you provide an example of why? And if not, have you tried using the non-`inout` version of `parse`?

In general we want to move in a direction that encourages using the non-`inout` version throughout the docs and tutorials.

This isn't to say we shouldn't provide an overload here should this PR land, though.

Reply:
Well… Mostly I was just copying your videos and playgrounds when trying out the parser library and so I wrote my tests like you guys did :)  

May also have been because of some of the non-`inout` `parse` functions weren't working on `Substring`?  I haven't had a chance to play with parsing lately, so I don't entirely recall.

Yes, I do have a couple of tests where I'm asserting that there is the expected output left after the parser runs.

I just switched all the other tests to use the non`input` version of `parse`, and they are _working fine_ now on this branch.

I took a quick scan through the referenced packages Sven listed below and I didn't see any there that require the `inout`.  It was quick, so I could have missed something, but wasn't obviously a big deal there.

So I think your sense that the `inout` version isn't needed as much appears to be spot on.